libplatsupport/CMakeLists.txt

```diff
@@ -34,10 +34,20 @@ config_choice(
     "ega;LibPlatSupportX86ConsoleDeviceEGA;LIB_PLAT_SUPPORT_SERIAL_TEXT_EGA;KernelPlatPC99"
 )
 
+if(NOT ${KernelArmMach} STREQUAL "")
+    set(LibPlatSupportMach "${KernelArmMach}")
+elseif(KernelPlatformRpi3)
+    set(LibPlatSupportMach "bcm283x")
+elseif(KernelPlatformRpi4)
+    set(LibPlatSupportMach "bcm283x")
+else()
+    set(LibPlatSupportMach "")
+endif()
+
 file(
     GLOB
         deps
-        src/mach/${KernelArmMach}/*.c
+        src/mach/${LibPlatSupportMach}/*.c
```

Comment:
If `LibPlatSupportMach` is empty this becomes `src/mach//*.c`. Seems this problem has also existed before already and this path is just ignored then. Should  be fixed one day. Can we have multiple file(GLOB ...) statement? Then we can simply do this
```
file(
    GLOB
        deps
        src/plat/${KernelPlatform}/*.c
        src/*.c
        src/plat/${KernelPlatform}/acpi/*.c
)

if(NOT ${LibPlatSupportMach} STREQUAL "")
    file(GLOB src/mach/${LibPlatSupportMach}/*.c)
endif()
````

Reply:
I was only able to make it work when I was first defining another GLOB variable since the * operator does not seem to work with list(APPEND deps src/mach/${LibPlatSupportMach}/*.c). I implemented something similar to https://stackoverflow.com/questions/7533502/how-can-i-merge-multiple-lists-of-files-together-with-cmake.